### ISSUE TITLE
refactor: remove unused access_token in validate CLI, load_reana_spec, workspaces

### ISSUE THREAD

**Issue by VMois** [reanahub/reana-client#567]
`workspace()` function doesn't need the `access_token` to work. We can remove `access_token` as a param from `workspces()` function, `validate` CLI, `load_reana_spec` function. 

One possible problem with it is that we will need to remove the `-t` option from the `workflow_validate` CLI which means a slight change of `reana-client` interface.

**Comment by mvidalgarcia** [comment]
In my opinion, the access token should be required. The access token is our only mechanism in `reana-client` to verify that a user is registered, I don't see any benefit to expose the available workspaces to non-registered users even if it's not sensitive information. What do you think?

**Comment by VMois** [comment]
I agree. From internal discussions, it was not clear for me whatever `access_token` is required for `workspaces` but I would prefer to have more security. If we all agree on that, we will need to modify the `reana-server` API and add a token to the `workspaces()` function in the `reana-client`.

@tiborsimko if you agree with that, I will create issues.

**Comment by tiborsimko** [comment]
(1) `reana-client workspaces` is returning only a general information about which workspace path prefixes the users may choose. It is not giving away the list of users workspaces, but based on the name, it can be easily misunderstood in that way. From this point of view, it should rather be renamed.

I think we can have a nice general `reana-client info` method which would return various "public" information about the instance. I.e. not only what workspace prefixes people can use, but also which workflow engines are supported, which compute backends are supported, what is the cluster version (as we do in "ping" command), and other such useful information to users.

This could work best if we return an "extensible" JSON response of key1-value1, key2-value2 so that we don't have to create any new endpoint should we want to disclose another public information in the future.

(2) I agree with @mvidalgarcia that it's best to keep the endpoint restricted to users who have tokens. This not only allows to restrict possibly sensitive information, but could come handy should we want to extend the information and include e.g. the number of running workflows, the number of pending workflows, etc as we already do in the status web page. Having this in the client could be handy e.g. for ATLAS pMSSM multi-workflow use cases.
